### DISCUSSION TITLE
[Needs Help] How to remove the old chunk files ?

### DISCUSSION THREAD

**Discussion by yeplaa** [fluent/fluentd#3798] [Q&A]
On an output type file, i would like that my old chunk files could be removing. Bellow my output configuration : 

    <match **kubernetes.var.log.**>
      @type file
      path /logs/fluentd
      time_slice_format %Y%m%d
      symlink_path /logs/fluentd/pipe
      <buffer>
        @type file_single
        path /logs/fluentd/buffer
        chunk_limit_size 32MB
        disable_chunk_backup true
        chunk_full_threshold 0.50
      </buffer>
    </match>

Bellow the files created in my directory : 

/logs $ ls -ltr
total 195508
-rw-r--r--    1 10000    10000     19296451 Jun 24 14:21 fluentd.20220624_0.log
-rw-r--r--    1 10000    10000     19131320 Jun 24 14:26 fluentd.20220624_1.log
-rw-r--r--    1 10000    10000     16965885 Jun 24 14:31 fluentd.20220624_2.log
-rw-r--r--    1 10000    10000     17302994 Jun 24 14:41 fluentd.20220624_3.log
-rw-r--r--    1 10000    10000     18702186 Jun 24 14:41 fluentd.20220624_4.log
-rw-r--r--    1 10000    10000     17447718 Jun 24 14:51 fluentd.20220624_5.log
-rw-r--r--    1 10000    10000     17044435 Jun 24 14:51 fluentd.20220624_6.log
-rw-r--r--    1 10000    10000     17911912 Jun 24 15:01 fluentd.20220624_7.log
-rw-r--r--    1 10000    10000     18638783 Jun 24 15:31 fluentd.20220624_8.log
-rw-r--r--    1 10000    10000     18864208 Jun 24 16:21 fluentd.20220624_9.log
-rw-r--r--    1 10000    10000     18864461 Jun 24 17:11 fluentd.20220624_10.log
drwxr-xr-x    3 10000    10000         4096 Jun 24 17:11 fluentd
/logs $ ls -ltr fluentd/
total 8
lrwxrwxrwx    1 10000    10000           63 Jun 24 17:11 pipe -> /logs/fluentd/buffer/fsb..b5e234a7e96797aa5fed13089ebfaf0b4.buf
drwxr-xr-x    2 10000    10000         4096 Jun 24 17:11 buffer

I would like to avoid that the fluentd.%Y%m%d.log files are created. Indeed, only the file linked to /logs/fluentd/pipe symbolik link is used.
I use the disable_chunk_backup feature but it doesn't work. Have you an idea? 

Thanks
Loïc

**Comment by fernandino143** [comment]
Has anyone ever looked at this? I'm facing the same issue now where "disable_chunk_backup" is not preventing the backup files from being created

**Comment by daipom** [comment]
> I would like to avoid that the fluentd.%Y%m%d.log files are created. Indeed, only the file linked to /logs/fluentd/pipe symbolik link is used.

This is not the intended use of the `out_file` plugin.
`out_file` plugin outputs files based on the `path` setting.
The `symlink_path` is just a buffering setting.

It depends on what you want to do, but perhaps you should consider using a different plugin.

> I use the disable_chunk_backup feature but it doesn't work.

>  I'm facing the same issue now where "disable_chunk_backup" is not preventing the backup files from being created

The `disable_chunk_backup` setting has nothing to do with this.
It is a setting related to backup in case of an anomaly.

* https://docs.fluentd.org/configuration/buffer-section
